### PR TITLE
add json title annotation

### PR DESCRIPTION
**Pull request by katlasik**
- Add title annotation for jsonschema

### CODE REVIEW

**Review by kciesielski — APPROVED** [review]

File: doc/docs/json-schema.md [hunk 1]
```diff
@@ -65,6 +66,7 @@ import sttp.tapir.generic.auto._
   val schemaStr: String = Printer.spaces2.print(schemaAsJson.deepDropNullValues)
 ```
 
+The title annotation of the object will be by default the name of the case class. You can customize it with `@title` annotation.
```
Comment (adamw) [inline doc/docs/json-schema.md:69]: hm do we want this to be the default? Is this somehow mandated by json schema, that the title is present? Maybe this should be an option of the serialiser, similar to `markOptionsAsNullable: Boolean`?

Comment (katlasik) [reply doc/docs/json-schema.md:69]: Sure, what do you propose as the name of the option?

Comment (adamw) [reply doc/docs/json-schema.md:69]: `addTitleToDefs`? :) I guess that's where the title is added.

Still an open question what should be the default. Maybe there are some examples of json schemas out there, for similar data structures, do they have the `title` set?

**Comment by katlasik** [comment]
closes https://github.com/softwaremill/sttp-apispec/issues/80

File: doc/docs/json-schema.md (hunk 1 above)
Comment (adamw) [reply doc/docs/json-schema.md:69]: Ok, let's set it by default to `true` when generating json-schema

**Pull request merged**